state/executor.go

```diff
@@ -414,6 +420,38 @@ func (t *Transition) nonceCheck(msg *types.Transaction) error {
 	return nil
 }
 
+// checkDynamicFees checks correctness of the EIP-1559 feature-related fields.
+// Basically, makes sure gas tip cap and gas fee cap are good.
+func (t *Transition) checkDynamicFees(msg *types.Transaction) error {
+	if msg.GasFeeCap.BitLen() == 0 && msg.GasTipCap.BitLen() == 0 {
```

Comment:
Is it correct to return nil here if it is a dynamic fee transaction? It's ok if it's a legacy transaction, but correct me if I'm wrong, if a EIP-1559 transaction has 0 as GasFeeCap, doesn't that mean that the user that sent the transaction is not willing to pay anything for it, which is not a valid use case?

Reply:
I'm not sure about that tho. In case when eip-1559 is not activated yet, users must be able to send transactions with empty dynamic fees. I can add a check to apply this validation only for dynamic tx when eip-1559 is activated already.

Reply:
Ok, I've analyzed the code a bit further. So if someone does set transaction type to `DynamicFeeTx`, but doesn't set the `GasFeeCap` and `GasTipCap`, or if they are zero, we will consider the given transaction as a legacy transaction? Am I correct? If that is the case, then this is ok.

Reply:
I think so. Do you think the behaviour should be different?

Reply:
To my knowledge, on `ethereum`, if you do not specify the the maximum fee you are willing to pay per unit of gas, your transaction gets rejected, but maybe we can put this in the RFC for further discussion.